### PR TITLE
formatting readme and adding more detailed intoramtion

### PR DESCRIPTION
**Pull request by AbdelrahmanDwedar**
There's kind of a lot of changes, I formatted the markdown following the best practices and also added more detailed stuff like the scripts for each command to make the customizing more flexable. Please let me know if there's any issue with the changes I made.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -53,6 +54,44 @@ Install with your preferred package manager:
...
+                    -- you can pass :map-arguments here
+                    opts = { desc = 'New line' },
+                },
+            },
+        }
+
+        keymap({ 'v', 'n' }, '<Leader>m', requires('multicursors').start, { desc = 'Create a selection for selcted text or word under the cursor' })
+    end,
+}
+```
```
Comment (smoka7) [inline README.md:93]:
```suggestion
```

File: README.md [hunk 2]
```diff
@@ -83,128 +122,132 @@ Install with your preferred package manager:
...
+| MCstart         | require("multicursors").start()              | Selects the word under cursor and starts listening for the actions. In visual mode it acts like `MCvisual` |
+| MCvisual        | require("multicursors").search_visual()      | Selects the last visual mode selection and starts listening for the actions.                               |
+| MCpattern       | require("multicursors").new_pattern()        | Prompts for a pattern and selects every match in the buffer.                                               |
+| MCvisualPattern | require("multicursors").new_pattern_visual() | Prompts for a pattern and selects every match in the visual selection.                                     |
+| MCunderCursor   | require("multicursors").new_under_cursor()   | Selects the char under cursor and starts listening for the actions.                                        |
+| MCclear         | require("multicursors").exit()               | Clears all the selection.                                                                                  |
```
Comment (smoka7) [inline README.md:137]:
```suggestion
```
The api isn't stable now, and i don't want to expose things that maybe gonna break in future.

File: README.md [hunk 3]
```diff
@@ -83,128 +122,132 @@ Install with your preferred package manager:
...
+| `Z`     | Aligns selections by adding space at beginning of line                          |
+| `d`     | Deletes the text inside selections                                              |
+| `D`     | `count` Deletes the text from start of selection till end of line               |
+| `dd`    | `count` Deletes line of selections                                              |
+| `@`     | Executes a macro at beginning of every selection                                |
+| `.`     | Reapets last change at the beginning of every selection                         |
+| `,`     | Clears All Selections except the main one                                       |
+| `:`     | Prompts for a normal command and Executes it at beginning of every selection    |
+| `u`     | Undo changes                                                                    |
+| `<C-r>` | Redo changes                                                                    |
```
Comment (smoka7) [inline README.md:183]:
```suggestion
```
It's unnecessary cause at the end all of it are aligned and every time we want to add a mapping or edit some description we have to edit all of them again

File: README.md [hunk 4]
```diff
@@ -83,128 +122,132 @@ Install with your preferred package manager:
...
+| `<C-Right>` | Moves the selections one word Right                  |
+| `<Home>`    | Moves the selections to start of line                |
+| `<End>`     | Moves the selections to end of line                  |
+| `<CR>`      | Insert one line below the selections                 |
+| `<C-j>`     | Insert one line below the selections                 |
+| `<C-v>`     | Pastes the text from system clipboard                |
+| `<C-r>`     | Insert the contents of a register                    |
+| `<C-w>`     | Deletes one word before the selections               |
+| `<C-BS>`    | Deletes one word before the selections               |
+| `<C-u>`     | Deletes froms start of selections till start of line |
```
Comment (smoka7) [inline README.md:212]:
```suggestion
```

File: README.md [hunk 5]
```diff
@@ -83,128 +122,132 @@ Install with your preferred package manager:
...
+| `h`     | `[count]` char left                                   |
+| `j`     | `[count]` char down                                   |
+| `k`     | `[count]` char up                                     |
+| `l`     | `[count]` char right                                  |
+| `t`     | Extends the selection to the parent of selected node  |
+| `r`     | Shrinks the selection to first child of selected node |
+| `y`     | Shrinks the selection to last child of selected node  |
+| `u`     | Undo Last selections extend or shrink                 |
+| `$`     | `[count]` to end of line                              |
+| `^`     | To the first non-blank character of the line          |
```
Comment (smoka7) [inline README.md:241]:
```suggestion
```

File: README.md [hunk 6]
```diff
@@ -7,7 +8,7 @@ The Multicursor Plugin for Neovim extends the native Neovim text editing capabil
 
 ## Installation
 
-Install with your preferred package manager:
+lazy.nim:
```
Comment (smoka7) [inline README.md:11]:
```suggestion
lazy.nvim:
```

**Comment by smoka7** [comment]
Thanks for your efforts

File: README.md (hunk 1 above)
Comment (AbdelrahmanDwedar) [reply README.md:93]: I think adding explanation for how to set it up with packer going to be a good idea. Also I still think that just the plugin and it's requires is enough for all plugin managers like so:
```lua
-- packer.nvim
use {
    'smoka7/multicursors.nvim',
    requires = {
        'nvim-treesitter/nvim-treesitter',
        'smoka7/hydra.nvim',
    }
}```

```lua
-- lazy.nvim
{
    "smoka7/multicursors.nvim",
    event = "VeryLazy",
    dependencies = {
        'nvim-treesitter/nvim-treesitter',
        'smoka7/hydra.nvim',
    },
}```

the other stuff we can move to another section for setup I think.

Comment (smoka7) [reply README.md:93]: Ok, I'll have to add some documents about how mappings work. I move it then.
I think your work with this pr is done. Thank you.

Comment (AbdelrahmanDwedar) [reply README.md:93]: No problem buddy.

Always open to help, this sounds like a great plugin I'd love to help with it.

**Pull request merged**